WhateverGreen/kern_igfx.hpp

```diff
@@ -499,6 +499,93 @@ class IGFX {
 		static bool forceWakeWaitAckFallback(uint32_t, uint32_t, uint32_t);
 		static void forceWake(void*, uint8_t set, uint32_t dom, uint32_t);
 	} ForceWakeWorkaround;
+	
+	/**
+	 *  Interface of a submodule to fix Intel graphics drivers
+	 */
+	class PatchSubmodule {
+	public:
+		/**
+		 *  Virtual destructor
+		 */
+		virtual ~PatchSubmodule() = default;
+		
+		/**
+		 *  True if this submodule should be enabled
+		 */
+		bool enabled {false};
```

Comment:
Why duplicate this with `available`? Just don't set `enabled` if you want to disable the patch.

Reply:
My intention is that this fix is available for ICL but can be backported to older platforms. As far as I know, older laptops normally have DVMT set to 32MB or 64MB, but it is possible that a weird one sets it to other values which are not supported by the formula. I left this `available` property on purpose to prevent users from misusing the patch as well. (Not all users will carefully read the manual.)

Reply:
So why can't you use the single variable, `enabled`, for exactly that purpose?

Reply:
Because they are split in time?

Reply:
Because `enabled` can be controlled by the boot argument or the device property, and then it is hard to know whether an user is trying to enable the patch on an unavailable platform.